components/dashboard/src/components/UsageBasedBillingConfig.tsx

```diff
@@ -90,14 +96,31 @@ export default function UsageBasedBillingConfig({ attributionId, hideSubheading
     }, [attributionId, refreshSubscriptionDetails]);
 
     useEffect(() => {
+        const params = new URLSearchParams(location.search);
+        setStripeParams({
+            setupIntentId: params.get("setup_intent") || undefined,
+            redirectStatus: params.get("redirect_status") || undefined,
+        });
+    }, [location.search]);
+
+    const subscribeToStripe = useCallback(() => {
         if (!attributionId) {
             return;
         }
-        const params = new URLSearchParams(location.search);
-        if (!params.get("setup_intent") || params.get("redirect_status") !== "succeeded") {
+        const { setupIntentId, redirectStatus } = stripeParams;
+        if (!setupIntentId || redirectStatus !== "succeeded") {
+            // TODO(gpl) We have to handle external validation errors (3DS, e.g.) here
             return;
         }
-        const setupIntentId = params.get("setup_intent")!;
+
+        // Guard against multiple execution following the pattern here: https://react.dev/learn/you-might-not-need-an-effect#initializing-the-application
+        if (didAlreadyCallSubscripe) {
```

Comment:
This is the only pattern I can find in the react-js docs for handling the pattern we have at hand :point_down: .

I agree with @AlexTugarev that the core issue is that we use Stripe Elements to re-direct into our own component, mixing triggers (and params). But digging into [their docs](https://stripe.com/docs/payments/quickstart?client=react), and also seeing how the integrate 3rd party authorization flows (by customer's banks UI is shown as full-screen overlay), I'm not sure how easy it is to translate that to alternatives (dedicated popup window, like we have for login, for instance).

Reply:
Definitely feels a bit odd, and maybe less needed with the useEffect suggestion I made above, but seems safe as well assuming a user can go through the flow again if needed.

If a user tries to go through the flow again, I suppose they'd have a page reload in between, so the flag would get initialized back to `false`, right?

Reply:
> If a user tries to go through the flow again, I suppose they'd have a page reload in between, so the flag would get initialized back to false, right?

That's my expectation, and what I saw in the testing I did, yes. :+1: